module/VuFind/src/VuFind/DigitalContent/OverdriveConnector.php

```diff
@@ -0,0 +1,819 @@
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\DigitalContent;
+
+use VuFind\View\Helper\Root\DateTime;
+use Zend\Log\LoggerAwareInterface;
+use Zend\Session\Container;
+use ZfcRbac\Service\AuthorizationServiceAwareInterface;
+use ZfcRbac\Service\AuthorizationServiceAwareTrait;
+
+
+
+/**
+ * OverdriveConnector
+ *
+ * Class responsible for connecting to the Overdrive API
+ *
+ * @category VuFind
+ * @package  AJAX
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Brent Palmer <brent-palmer@icpl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ * @todo provide option for autocheckout by default in config
+ *       allow override for cover display using Overdrive covers
+ *       provide option for not requiring email for holds
+ *       provide option for giving users option for every hold
+ *       provide option for asking about autocheckout for every hold
+ *       provide config options for how to handle patrons with no access to OD
+ *       look into storing collection token in application (object) cache
+ *         instead of the users session.
+ 
+ */
+class OverdriveConnector implements LoggerAwareInterface, 
+     AuthorizationServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+    use AuthorizationServiceAwareTrait;
+    
+    /**
+     * Session Container
+     * desc
+     * @var string
+     */
+    protected $sessionContainer;
+
+    /**
+     * desc
+     * @var string
+     */
+    protected $recordConfig;
+
+    /**
+     * desc
+     * @var string
+     */
+    protected $mainConfig;
+
+    /**
+     * desc
+     * @var string
+     */
+    protected $ilsAuth;
+    
+    /**
+     * Constructor
+     *
+     * @param \Zend\Config\Config $recordConfig   VuFind main configuration (omit for
+     *                                            built-in defaults)
+     *
+     * @param \Zend\Config\Config $recordConfig   Record-specific configuration file
+     *                                            (omit to use $recordConfig as $recordConfig)
+     * @param \Zend\Config\Config $searchSettings Search-specific configuration file
+     * @param Callable            $sessionFactory Factory function returning SessionContainer object
+     */
+    public function __construct(
+        $mainConfig = null,
+        $recordConfig = null,
+        $sessManager,
+        $ilsAuth
+    ) {
+        $this->debug("SolrOverdrive Connector");
+        $this->mainConfig = $mainConfig;
+        $this->recordConfig = $recordConfig;
+        $this->ilsAuth = $ilsAuth;
+        
+        
+        // Init session cache for session-specific data
+        if ($conf=$this->getConfig()) {
+            $namespace = md5("VuFind\DigitalContent");
+            $this->sessionContainer = new Container($namespace);
```

Comment:
When creating a session container, you need to inject the session manager -- I see that you are passing it as a parameter to the constructor, but it doesn't seem to be getting used here.

Reply:
I don't think I needed to inject the session manager at all.  I just instantiate a new session container and start using it.  For now I have removed it from the constructor call.  Let me know if you think I've missed something.

Reply:
Zend\Session\Container takes a session manager as the second constructor parameter. If you don't inject it, it creates its own -- but that causes problems, because then you end up with two independent, conflicting session managers in use -- the one VuFind created, and the one the container created. This sometimes causes weird, subtle bugs. It's definitely a good idea to inject the manager, though the code will at least superficially appear to work if you do not.